src/plugins/BehaviorSpecGenerator/BehaviorSpecGenerator.js

```diff
@@ -254,28 +267,237 @@ define([
                 if (componentTypeNames.hasOwnProperty(name)) {
                     violations.push({
                         node: node,
-                        message: 'Duplicated name [' + name + '] shared with ' + componentTypeNames[name]
+                        message: 'Duplicated component [' + name + '] shared with ' + componentTypeNames[name]
                     });
                 }
                 componentTypeNames[name] = this.core.getPath(node);
 
+                var constructors = this.core.getAttribute(node, 'constructors');
+                if (constructors == '') {
+                    violations.push({
+                        node: node,
+                        message: 'Constructors empty for' + componentTypeNames[name]
+                    });
+                }
+                var definitions = this.core.getAttribute(node, 'definitions');
+                if (definitions == '') {
+                    violations.push({
+                        node: node,
+                        message: 'Definitions empty for' + componentTypeNames[name]
+                    });
+                }
+                var forwards = this.core.getAttribute(node, 'forwards');
+                if (definitions == '') {
+                    violations.push({
+                        node: node,
+                        message: 'forwards empty for' + componentTypeNames[name]
+                    });
+                }
+
                 // check for states,guards and transitions in each componentType
 
-                /*for (var child of this.core.getChildrenPaths(node))
-                {
+                for (var childPath of this.core.getChildrenPaths(node)) {
+                    var child = nodes[childPath];
+                    var childName = this.core.getAttribute(child, 'name');
+
                     if ((this.isMetaTypeOf(child, this.META.State)) || (this.isMetaTypeOf(child, this.META.InitialState))) {
-                        if (stateNames.hasOwnProperty(name)) {
+                        if (totalStateNames.hasOwnProperty(childName)) {
+                            violations.push({
+                                node: node,
+                                message: 'Duplicated State [' + childName + '] shared with ' + totalStateNames[childName]
+                            });
+                        }
+                        totalStateNames[childName] = this.core.getPath(child);
+                    }
+
+                    if ((this.isMetaTypeOf(child, this.META.EnforceableTransition)) || (this.isMetaTypeOf(child, this.META.SpontaneousTransition))) {
+
+                        if (this.core.getPointerPath(child, 'dst') === null) {
+                            violations.push({
+                                    node: node,
+                                message: 'Dst of connector [' + childPath + '] is null'
```

Comment:
`Connection, <nameOfConnection>, with no destination encountered. Connect or remove it.`